src/Compilers/CSharp/Portable/BoundTree/BoundExpression.cs

```diff
@@ -19,7 +19,12 @@ public SimpleNameSyntax? InterceptableNameSyntax
             {
                 // When this assertion fails, it means a new syntax is being used which corresponds to a BoundCall.
                 // The developer needs to determine how this new syntax should interact with interceptors (produce an error, permit intercepting the call, etc...)
-                Debug.Assert(this.WasCompilerGenerated || this.Syntax is InvocationExpressionSyntax or ConstructorInitializerSyntax or PrimaryConstructorBaseTypeSyntax { ArgumentList: { } },
+                Debug.Assert(this.WasCompilerGenerated ||
+                    this.Syntax is
+                        InvocationExpressionSyntax or
+                        ConstructorInitializerSyntax or
+                        PrimaryConstructorBaseTypeSyntax { ArgumentList: { } } or
+                        WithElementSyntax,
```

Comment:
Do we need a test for this scenario? (interceptor intentionally cannot affect synthesized `Create` call)